### ISSUE TITLE
Replace unnecessary calls to ``.flatten()`` with ``.ravel()``.

### ISSUE THREAD

**Issue by rileyjmurray** [sandialabs/pyGSTi#431]
As of the latest commit on [develop](https://github.com/sandialabs/pyGSTi/tree/ad0c1a5cbcdf081419d8b68ded0ae0da067da66b) (between releases of pyGSTi 0.9.12 and 0.9.13)m there are 102 hits for ``.flatten()`` in the pyGSTi codebase.

The [documentation for flatten](https://numpy.org/doc/1.25/reference/generated/numpy.ndarray.flatten.html) says that this returns a flattened copy of the input array. In the vast majority of situations it's preferable to use [``.ravel()``](https://numpy.org/doc/1.25/reference/generated/numpy.ravel.html#numpy.ravel) instead, which only returns a copy when necessary. I'll make a PR in the near future for these changes.

This is spiritually similar to #429 and #430, but different enough to warrant its own tracking issue.

**Comment by coreyostrove** [comment]
Sounds great!

**Comment by pcwysoc** [comment]
Could you explain a bit more when .ravel() can be used instead of .flatten()? I'm confused about when a copy is necessary.

**Comment by rileyjmurray** [comment]
@pcwysoc, as a rule of thumb, copies _aren't_ necessary. Python (or rather, numpy) makes them all the time. The main time when a copy would be important is if you're explicitly using in-place operations or if you're overwriting portions of an array.
```python
import numpy as np

a = np.ones((3,5))
b = a.ravel()
b = b + 2
print(a) # unchanged, because the change to b1 wasn't specifically in-place.

b = a.ravel()
b += 2
print(a) # changed, since "+=" is an in-place operation

a = np.ones((3,5))
b = a.ravel()
b[::2] = 0.0  # overwrite every other element of b with 0.0.
print(a) # changed, since b shares memory with a
```

**Comment by sserita** [comment]
Fixed in #451
